config/config.go

```diff
@@ -117,6 +117,9 @@ type AnalyzersList struct {
 
 // SourceConfig has some controls about what chain we're analyzing and how to connect.
 type SourceConfig struct {
+	// Cache holds the configuration for a file-based caching backend.
+	Cache *CacheList `koanf:"cache"`
```

Comment:
Do you have a use case in mind that benefits from having separate configs for consensus and runtime? I can't, and would suggest a single `Cache *CacheConfig` here.

Even if they ever needed to be separate, I think it's more likely we'd want to have per-runtime control, e.g. "cache sapphire but not emerald or consensus". But let's not go there.

Reply:
Ah I was actually going to move further in the direction of per-runtime configurations. I think it'll be useful to store the consensus data and runtime(s) data separately. That way, changes to the consensus or runtime analyzers would only require updating the corresponding dataset. And if we do end up using the file-based indexer to speed up full reindexes, it'll save us from having to re-pull a lot of data from node.

An alternative would be to just list out all the cache directories in a single `CacheConfig`, and have a single `QueryOnCacheMiss` flag.

Reply:
> That way, changes to the consensus or runtime analyzers would only require updating the corresponding dataset.

Hopefully, changes to analyzers will not require any changes to the cached dataset, because we're caching the _input_ to the analyzers. The cache might need to be expanded with new results of new RPCs, but there's no benefit to per-runtime configs there. Or, in the rare case, the NodeApiLite might need to start producing a differently-normalized version of the node response. In that case, we might want to throw away some cache keys, but again at a different granularity than per-runtime. If that happens, we can still modify the FileApiLite layer so it updates the cached entry if it's not using the newest format.

The suggestion in your last paragraph is very close to my suggestion actually. And there is potential value to having separate DBs per runtime. Let's go with your last paragraph; I would just further suggest to simplify a tiny bit more and take a single directory path in the config, then auto-derive subfolders in the FileApiLite impls. So config would say something like `/data/oasis-indexer/rpc-cache`, and DBs would be created in `/data/oasis-indexer/rpc-cache/consensus`, `/data/oasis-indexer/rpc-cache/emerald`, itd.